lib/matplotlib/blocking_input.py

```diff
@@ -2,133 +2,111 @@
 
 import six
-from matplotlib import verbose
-import matplotlib.lines as mlines
 
+from numbers import Integral
 import warnings
 
+from matplotlib import verbose
+import matplotlib.lines as mlines
+import numpy as np
+
 
 class BlockingInput(object):
-    """
-    Class that creates a callable object to retrieve events in a
-    blocking way.
+    """Callable for retrieving events in a blocking way.
     """
     def __init__(self, fig, eventslist=()):
         self.fig = fig
         self.eventslist = eventslist
 
     def on_event(self, event):
         """
-        Event handler that will be passed to the current figure to
-        retrieve events.
+        Event handler; will be passed to the current figure to retrieve events.
         """
-        # Add a new event to list - using a separate function is
-        # overkill for the base class, but this is consistent with
-        # subclasses
+        # Add a new event to list - using a separate function is overkill for
+        # the base class, but this is consistent with subclasses.
         self.add_event(event)
 
         verbose.report("Event %i" % len(self.events))
 
-        # This will extract info from events
+        # This will extract info from events.
         self.post_event()
 
-        # Check if we have enough events already
+        # Check if we have enough events already.
         if len(self.events) >= self.n and self.n > 0:
             self.fig.canvas.stop_event_loop()
 
     def post_event(self):
-        """For baseclass, do nothing but collect events"""
-        pass
+        """For baseclass, do nothing but collect events."""
 
     def cleanup(self):
-        """Disconnect all callbacks"""
+        """Disconnect all callbacks."""
         for cb in self.callbacks:
             self.fig.canvas.mpl_disconnect(cb)
-
         self.callbacks = []
 
     def add_event(self, event):
         """For base class, this just appends an event to events."""
         self.events.append(event)
 
     def pop_event(self, index=-1):
-        """
-        This removes an event from the event list.  Defaults to
-        removing last event, but an index can be supplied.  Note that
-        this does not check that there are events, much like the
-        normal pop method.  If not events exist, this will throw an
-        exception.
+        """Remove an event from the event list -- by default, the last.
+
+        Note that this does not check that there are events, much like the
+        normal pop method.  If no events exist, this will throw an exception.
         """
         self.events.pop(index)
 
-    def pop(self, index=-1):
-        self.pop_event(index)
-    pop.__doc__ = pop_event.__doc__
+    pop = pop_event
 
     def __call__(self, n=1, timeout=30):
-        """
-        Blocking call to retrieve n events
-        """
-
-        if not isinstance(n, int):
+        """Blocking call to retrieve *n* events."""
+        if not isinstance(n, (Integral, np.integer)):
```

Comment:
allows one to pass a numpy integer rather than just a python builtin integer

Reply:
Thanks - my brain wasn't parsing `np.int` as different than `integer`